### ISSUE TITLE
firebase_crashlytics errors not arriving at firebase

### ISSUE THREAD

**Issue by safield** [FirebaseExtended/flutterfire#2846]
I am using the firebase_crashlytics lib in my Flutter app.

I am sending error reports, but only about 1 in 100 arrive.....and I can not discern the difference between the ones that arrive, and the ones that don't.

All the code pertaining to crashlytics is as follows....

    static void asrt(bool condition)
    {
        if (condition)
            return;

        Crashlytics.instance.enableInDevMode = true;
        StringBuffer sb = StringBuffer();

        for (var logLine in _getInstance()._logTailBuffer)
            sb.writeln(logLine);

        try {
            throw 0;
        }
        catch (error , stacktrace) {
            String tempIdentifier =  stacktrace.toString().substring(0,160);
            Crashlytics.instance.log(sb.toString());
            Crashlytics.instance.recordError(niceIdentifier , stacktrace);
        }
    }

flutter logs shows the following lines when a report is sent..

```
I/flutter ( 8060): Error caught by Crashlytics plugin <recordError>:
I/flutter ( 8060): firebase_crashlytics: Error reported to Crashlytics.
```

**Comment by kanhaiyhacker** [comment]
In the firebase crashlytics console section change the event type to non-fatal as by default event type is crashes. In flutter app won't crash and just it just though error.

**Comment by TahaTesser** [comment]
Hi @safield 
Looks like duplicate of https://github.com/FirebaseExtended/flutterfire/issues/2644 
Are you experiencing same issue?
Thank you

**Comment by safield** [comment]
The issue was caused by a funny series of events. I just learned that the error report is not sent until the application is opened a second time after the crash. What was happening is that I would launch the app and trigger a crash, then observe that no crash report arrived. I would then try changing some code to see if I could change the results. This would result in a new apk being installed, and so the pending crash report was never sent.

I guess I was reliably always installing a new apk after each crash. 

It would be nice if  it was sent immediately (like sentry does).
